plane/src/cleanup.rs

```diff
@@ -0,0 +1,36 @@
+use crate::database::PlaneDatabase;
+use anyhow::Result;
+
+const CLEANUP_LOOP_INTERVAL_SECONDS: u64 = 60 * 15;
+
+pub async fn run_cleanup(db: &PlaneDatabase, min_age_days: Option<i32>) -> Result<()> {
+    tracing::info!("Running cleanup");
+
+    if let Some(min_age_days) = min_age_days {
+        db.backend().cleanup(min_age_days).await?;
+    }
+
+    db.clean_up_tokens().await?;
+
+    tracing::info!("Done running cleanup");
+
+    Ok(())
+}
+
+pub async fn run_cleanup_loop(db: PlaneDatabase, min_age_days: Option<i32>) {
+    // Each controller runs a cleanup loop. To avoid having them all run at the same time, we
+    // introduce a random offset to the start time.
```

Comment:
So if we have 3 controllers running, does that mean we'll be cleaning up the DB every 5 minutes on average?

Reply:
Yep. There is absolutely nothing that tries to make the time gap uniform, but I think that's fine. These should be pretty fast queries.